File: Day 8/day8.py
def draw_race(indices: list[int], length: int) -> str:
    n = len(indices)
    resultado = []

    for i, idx in enumerate(indices):
        carril = '~' * length

        if idx > 0:
            carril = '~'*idx + 'r' + '~'*(length-idx-1)
        elif idx < 0:
            idx_n = length + idx
            carril = '~'*idx_n + 'r' + '~'*(-idx-1)

        fi_carr = ' /' + str(i+1)
        resultado.append(' '*(n - i+1) + carril + fi_carr)

    return '\n'.join(resultado)

File: Day 8/day8.ts
function drawRace(indices: number[], length: number): string{
    const n = indices.length;
    const resultado: string[] = [];

    for (let i = 0; i < n; i++){
        const idx = indices[i]
        let carril = '~'.repeat(length)

        if (idx > 0) {carril = '~'.repeat(idx) + 'r' + '~'.repeat(length-idx-1);}
        else if (idx < 0) {
            const idxN = length+idx;
            carril = '~'.repeat(idxN) + 'r' + '~'.repeat(-idx-1);
        }

        const fiCarr = ' /' + (i+1).toString();
        resultado.push(' '.repeat(n - i+1) + carril + fiCarr);
    }


    return resultado.join("\n");
}